backend/app.py

```
@@ -375,3 +315,36 @@ async def handle_mcp_streamable_http(request: Request) -> Response:
+        # Check if it's a static file request (has file extension in last path segment)
+        path_parts = full_path.split("/")
+        if path_parts and "." in path_parts[-1]:
+            # Try to serve as static file
+            file_path = _static_path / full_path
+            if file_path.exists() and file_path.is_file():
+                return FileResponse(file_path)
+        # For all other routes (no extension or file not found), serve index.html
+        # React Router will handle the routing client-side
+        return FileResponse(_index_path)
+
+    LOGGER.info(f"Static files and SPA routing configured from {_static_path}")
```

Ruff (0.14.7)

Logging statement uses f-string

(G004)